adapters/tlsio_appleios.c

```diff
@@ -0,0 +1,508 @@
+                }
+                result = 0;
+                send_size = 0;
+            }
+            else /* Still have buffer to transmit. */
+            {
+                runBuffer += send_result;
+                send_size -= send_result;
+            }
+        }
+    }
+    return result;
+}
+
+/* Codes_SRS_TLSIO_APPLEIOS_32_062: [ The tlsio_appleios_dowork shall execute the async jobs for the tlsio. ]*/
+void tlsio_appleios_dowork(CONCRETE_IO_HANDLE tlsio_handle)
+{
+    if (tlsio_handle == NULL)
+    {
+        /* Codes_SRS_TLSIO_APPLEIOS_32_074: [ If the tlsio_handle is NULL, the tlsio_appleios_dowork shall not do anything. ]*/
+        LogError("Invalid parameter");
+    }
+    else
+    {
+        long received;
+        TLS_IO_INSTANCE* tlsio_instance = (TLS_IO_INSTANCE*)tlsio_handle;
+        /* Codes_SRS_TLSIO_APPLEIOS_32_075: [ The tlsio_appleios_dowork shall create a buffer to store the data received from the ssl client. ]*/
+        /* Codes_SRS_TLSIO_APPLEIOS_32_076: [ The tlsio_appleios_dowork shall delete the buffer to store the data received from the ssl client. ]*/
+        uint8_t RecvBuffer[RECEIVE_BUFFER_SIZE];
+		
+	    CFStreamStatus statusRead = CFReadStreamGetStatus(tlsio_instance->sockRead);
+	    //CFStreamStatus statusWrite = CFWriteStreamGetStatus(tlsio_instance->sockWrite);
+
+        switch (tlsio_instance->state)
+        {
+        case TLSIO_APPLEIOS_STATE_OPENING:
+            if (statusRead == kCFStreamStatusOpen)
+            {
+                /* Codes_SRS_TLSIO_APPLEIOS_32_063: [ If the tlsio state is TLSIO_APPLEIOS_STATE_OPENING, and ssl client is connected, the tlsio_appleios_dowork shall change the tlsio state to TLSIO_APPLEIOS_STATE_OPEN, and call the on_io_open_complete with IO_OPEN_OK. ]*/
+                tlsio_instance->state = TLSIO_APPLEIOS_STATE_OPEN;
+                /* Codes_SRS_TLSIO_APPLEIOS_32_002: [ The tlsio_appleios shall report the open operation status using the IO_OPEN_RESULT enumerator defined in the `xio.h`.]*/
+                CallOpenCallback(IO_OPEN_OK);
+            }
+            break;
+        case TLSIO_APPLEIOS_STATE_OPEN:
+            if (statusRead == kCFStreamStatusNotOpen || statusRead == kCFStreamStatusError)
+            {
+                /* Codes_SRS_TLSIO_APPLEIOS_32_071: [ If the tlsio state is TLSIO_APPLEIOS_STATE_OPEN, and ssl client is not connected, the tlsio_appleios_dowork shall change the state to TLSIO_APPLEIOS_STATE_ERROR, call on_io_error. ]*/
+                tlsio_instance->state = TLSIO_APPLEIOS_STATE_ERROR;
+                LogError("SSL closed the connection.");
+                CallErrorCallback();
+            }
+            else
+            {
+                /* Codes_SRS_TLSIO_APPLEIOS_32_069: [ If the tlsio state is TLSIO_APPLEIOS_STATE_OPEN, the tlsio_appleios_dowork shall read data from the ssl client. ]*/
+				while (CFReadStreamHasBytesAvailable(tlsio_instance->sockRead))
```

Comment:
Please add a limit on this loop, for instance, a maximum of 10 reads per dowork call.